Refactor/crud

Gonzom opened the pull request:
Creating a version 2 system of the current user and event schemas and models and connecting them with CRUD operations.
Tests are included.

The version 2 is not integrated in the system and is segregated to allow easier implementation.

Gonzom committed:
refactor: created crud entry points

Creating a version 2 system of the current user and event schemas and models and connecting them with CRUD operations. Tests are included.

The version 2 is not integrated in the system and is segregated.

Gonzom committed:
refactor: moved locales path to dependencies.py

Updated relevant code and tests.

yammesicka commented on app/database/models_v2.py:
```diff
@@ -0,0 +1,310 @@
+        nullable=False,
+        doc=PRIMARY_KEY_DOC,
+    )
+    # category_id = Column(
+    #     Integer,
+    #     ForeignKey("categories.id"),
+    #     nullable=False,
+    #     doc="",  # TODO: doc
+    # )
```
Should we add it or not? :)

Gonzom replied:
I haven't done all classes, but probably better to remove commented out and re-add it later.

Gonzom replied:
Done.

yammesicka commented on app/database/models_v2.py:
```diff
@@ -0,0 +1,310 @@
+"""SQLAlchemy database models."""
```
I think `models.py` is a good name. If there's already `models.py` file, you can `git move` it to `models_old.py`

yammesicka commented on app/database/schemas_v2.py:
```diff
@@ -0,0 +1,305 @@
+class EventBase(BaseModel):
+    """Base Event schema model. Should not be used directly."""
+
+    title: Annotated[
```
I think we might want to keep supporting Python 3.8 to allow Ubuntu 20.04 LTS support our system OOTB

Gonzom replied:
Done.

yammesicka commented on app/internal/languages.py:
```diff
@@ -34,14 +32,12 @@ def set_ui_language(language: str = None) -> None:
-    language_dir = _get_language_directory()
-
-    if language not in list(_get_supported_languages(language_dir)):
+    if language not in list(_get_supported_languages()):
```
Maybe `set(_get_supported_languages)` to get faster results?

You can also omit the casting to `list` if the performance is not an issue

Gonzom replied:
Probably misunderstood you here, so would appreciate further clarification:
```
>       if language not in set(_get_supported_languages()):
E       TypeError: unhashable type: 'list
```

Gonzom replied:
fixed.

yammesicka commented on app/internal/languages.py:
```diff
@@ -59,27 +55,7 @@ def set_ui_language(language: str = None) -> None:
-    paths = [Path(f.path) for f in os.scandir(language_dir) if f.is_dir()]
+def _get_supported_languages() -> Iterator[str]:
+    """Returns a generator of supported translation languages codes."""
+    paths = [Path(f.path) for f in os.scandir(LOCALES_PATH) if f.is_dir()]
```
I'd go with generator expression here (so the memory won't store many paths in a single time)

Gonzom replied:
Isn't it now?
```
    paths = [Path(f.path) for f in os.scandir(LOCALES_PATH) if f.is_dir()]
    return (language.name for language in paths)
```

Gonzom replied:
fixed.

yammesicka reviewed (commented):
Awesome work!! Great code, well tested. Kudos

Gonzom committed:
fix: added a constraint to the user_event

This constraint makes it so a user can only be added once to an event.

Gonzom committed:
test: added and fixed tests

* Added missing test for base attributes.
* Renamed get_set tests to only get_, as they don't preform any setting of values.
* Fixed a bug with test_get_change_owner()
* Added an additional test to test_get_add_delete_members() to check that the unique constraint works.
* Added new util function.

Gonzom committed:
fix: fixed logic in change_owner()

Gonzom committed:
fix: remove commented out code

Gonzom committed:
refactor: removed Annotated usage per PR request

Gonzom committed:
refactor: moved User model to correct alphabetical order

Gonzom committed:
refactor: changed attribute name

This change better reflects the data it represents as the data is not a date object, but a complete datetime object.

Gonzom committed:
fix: CR fixes
